python/cvmfs/repository.py

```diff
@@ -197,39 +196,230 @@ def _pop_catalog(self):
         return wrapper.get_catalog()
 
 
+class Resource(file):
```

Comment:
Even though I like this [RAII](http://www.wikiwand.com/en/Resource_Acquisition_Is_Initialization) idea, I think it's overkill here and kind of adds unclear magic. Looking at `retrieve_file()` and `retrieve_raw_file()` it is completely unclear how the control flow happens on first glance. See there please.